packages/opencensus-core/src/trace/model/tracer-base.ts

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as uuid from 'uuid';
+import * as logger from '../../common/console-logger';
+import * as loggerTypes from '../../common/types';
+import * as configTypes from '../config/types';
+import {TraceParams} from '../config/types';
+import {noopPropagation} from '../propagation/noop-propagation';
+import {Propagation} from '../propagation/types';
+import {DEFAULT_SAMPLING_RATE, SamplerBuilder, TraceParamsBuilder} from '../sampler/sampler';
+import * as samplerTypes from '../sampler/types';
+import {NoRecordRootSpan} from './no-record/no-record-root-span';
+import {NoRecordSpan} from './no-record/no-record-span';
+import {RootSpan} from './root-span';
+import * as types from './types';
+
+/**
+ * This class represents a tracer.
+ */
+export class CoreTracerBase implements types.TracerBase {
+  /** Indicates if the tracer is active */
+  private activeLocal: boolean;
+  /** A configuration for starting the tracer */
+  private config!: configTypes.TracerConfig;
```

Comment:
Rather than using a `!` would it be possible to assign some reasonable default config?

Reply:
The default config leaves in the `opencensus-node` today. I could move it to `opencensus-core` and add as a static property to this class. Then use it here as defaults and consume from here in the `opencensus-node`. My only concern is that this is not a change in this PR and I would prefer to keep the scope focused. But I'm happy to do this as a separate PR.